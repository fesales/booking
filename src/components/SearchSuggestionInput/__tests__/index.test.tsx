import React from 'react';
import { shallow } from 'enzyme';
import SearchSuggestionsInput from '..';

describe('SearchSuggestionsInput', () => {
  const defaultProps = {
    label: 'input label',
    placeholder: 'placeholder',
    onSearchTermChanged: () => {},
  };

  it('should render component', () => {
    const wrapper = shallow(<SearchSuggestionsInput {...defaultProps} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('form')).toHaveLength(1);
    expect(wrapper.find('input')).toHaveLength(1);
    expect(wrapper.find('input').prop('placeholder')).toEqual(defaultProps.placeholder);
    expect(wrapper.find('label')).toHaveLength(1);
    expect(wrapper.find('label').text()).toEqual(defaultProps.label);
  });

  it('should trigger search term changed with input value', () => {
    const props = {
      ...defaultProps,
      onSearchTermChanged: jest.fn(),
    };
    const mockedEvent = {
      target: {
        value: 'search term',
      },
    };

    const wrapper = shallow(<SearchSuggestionsInput {...props} />);

    wrapper.find('input').simulate('change', mockedEvent);

    expect(props.onSearchTermChanged).toHaveBeenCalledTimes(1);
    expect(props.onSearchTermChanged).toHaveBeenCalledWith(mockedEvent.target.value);
  });
});
