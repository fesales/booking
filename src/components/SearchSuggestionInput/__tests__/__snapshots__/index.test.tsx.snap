// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSuggestionsInput should render component 1`] = `
<form>
  <label
    className="search-input-label"
    htmlFor="search-input"
  >
    input label
  </label>
  <input
    className="search-input"
    name="search-input"
    onChange={[Function]}
    placeholder="placeholder"
    type="text"
  />
</form>
`;
